components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1116,7 +1155,7 @@ where
         let term = entry.get_term();
         let data = entry.get_data();
 
-        if !data.is_empty() {
+        if (self.peer.is_witness && !can_witness_skip(entry)) || !data.is_empty() {
```

Comment:
If it's witness before, but a command changes it to non-witness, and starts applying all following command. It can cause problem if the command depends on previous side effect.

Reply:
When it changes to non-witness, it should be marked with `unavailable` and stop applying the following commands until the snapshot is applied. This is considered in next PR.

Reply:
It may be conflict with existing mechanism. Currently, applying snapshot requires applied index is updated to last scheduled index. If applying is stopped, applying snapshot is also blocked.

Reply:
Add a note to consider `ready_to_handle_pending_snap` in https://github.com/tikv/tikv/pull/13491